s3/replication/manager/src/s3replicationmanager/app.py

```diff
@@ -17,11 +17,176 @@
+            return web.json_response({'Response': response})
+        else:
+            return web.json_response({'Response': 'INVALID subscriber'})
+
+    else:
+        return web.json_response({'jobs': list(jobs.keys())})
+
+@routes.get('/jobs/{job_id}')
+async def get_job(request):
+    """Get job attribute
+
+    Handler api to fetch job attributes
+
+    """
+    id = request.match_info['job_id']
+
+    # Get job's attribute
+    if id in jobs.keys():
+        return web.json_response(jobs[id])
+    else:
+        return web.json_response({'Response': 'ERROR : Job is not present!'})
+
+@routes.post('/jobs')
+async def add_job(request):
+    """Add jobs
+
+    Handler to add jobs to the job queue
+
+    """
+    entries = await request.json()
+    # Add job to the list
+    jobs.update(entries)
+    LOG.debug('Job list : {}'.format(jobs))
+    return web.json_response({'Response': 'jobs added!'})
+
+@routes.post('/subscribers')
+async def add_subscriber(request):
+    """Add subscriber
+
+    Handler for Subscriber
+
+    """
+    subscriber = await request.json()
+    LOG.debug('subscriber is : {}'.format(subscriber.get('sub_id')))
+    sub_id = subscriber.get('sub_id')
+
+    # Check if subscriber is already present
+    if sub_id in subscribers:
+        return web.json_response({'Response': 'Replicator is Already subscribed!'})
+    else:
+        subscribers.append(subscriber.get('sub_id'))
+        LOG.debug(subscribers)
+        return web.json_response({'subscribers': subscribers})
+
+@routes.get('/subscribers')
+async def list_subscribers(request):
+    """List subscriber
+
+    Handler to get subscriber list
+
+    """
+    return web.json_response({'subscribers': subscribers})
+
+@routes.put('/jobs/{job_id}')
+async def update_job_attr(request):
+    """Update job attributes
+
+    Update attributes for job_id
+
+    """
+    val = await request.json()
+    id = (request.match_info['job_id'])
+    LOG.debug('id is {}'.format(id))
+
+    # Check if key is present
+    if id in jobs.keys():
+        jobs[id] = val
+        return web.json_response({id: jobs[id]})
+    else:
+        jobs.update({id: val})
+        # jobs_inprogress.append(id)
+        return web.json_response({'jobs': jobs})
 
 class ReplicationManagerApp:
-  def __init__(self):
-    """Initialise logger and configuration."""
+    def __init__(self, configfile):
+        """Initialise logger and configuration."""
+
+        confReplicator = Config(configfile)
+
+        HOST = confReplicator.host
+        PORT = confReplicator.port
+
+        # setup logging
+        setup_logging()
+
+        LOG.debug('HOST is : {}'.format(HOST))
+        LOG.debug('PORT is : {}'.format(PORT))
 
-  def run(self):
-    Logger().print("Started replication manager using common logger...")
\ No newline at end of file
+    def run(self):
+        app = web.Application()
+        app.add_routes(routes)
+        web.run_app(app, host=HOST, port=PORT)
```

Comment:
with self.config this can change to 

web.run_app(app, host=self.config.get_host(), port=self.config.get_port())

Reply:
self.conf.host and self.conf.post used instead of get method.